main/src/main/scala/sbt/internal/RemoteCache.scala

```diff
@@ -0,0 +1,217 @@
+package sbt
+package internal
+
+import java.io.File
+import Keys._
+import SlashSyntax0._
+import Project._ // for tag and inTask()
+import std.TaskExtra._ // for join
+import sbt.coursierint.LMCoursier
+import sbt.librarymanagement._
+import sbt.librarymanagement.ivy.Credentials
+import sbt.librarymanagement.syntax._
+import sbt.internal.librarymanagement._
+import sbt.io.IO
+import sbt.io.syntax._
+import sbt.internal.inc.JarUtils
+
+object RemoteCache {
+  final val cachedCompileClassifier = "cached-compile"
+  final val cachedTestClasifier = "cached-test"
+
+  def gitCommitId: String =
+    scala.sys.process.Process("git rev-parse --short HEAD").!!.trim
```

Comment:
I think that this line introduce a dependency from the git command line tool.
You can possibly do the same using jgit:
```java
git.getRepository().exactRef("refs/heads/master").getObjectId().abbreviate(7).name()
```
from:
https://stackoverflow.com/questions/39356463/is-there-a-jgit-method-equivalent-to-git-rev-parse-short

It might be an additional dependency for Sbt itself, but I think it would be more "self-contained", what do you think?

Reply:
Can this be abstract over and provides an implemented with git remote cache out of the box?

Reply:
And I don't think it's a good idea to use git as a file cache.

Reply:
I'm of double mind here. On one hand you're right that it would be good to be self-contained. But on the other, every layer of J-anything could lag behind and introduce problem on its own. See for example `pgp` vs BouncyCastle. If ppl want JGit support they could use sbt-git and write

```scala
ThisBuild / git.gitRunner := com.typesafe.sbt.git.JGitRunner
ThisBuild / remoteCacheId := git.gitHeadCommit.value
```

or something to the effect?

Reply:
> And I don't think it's a good idea to use git as a file cache.

The git commit id is the one for the source code, file cache will be a regular Maven repo.

Reply:
I understand your concerns, still, for exactly one call I would prefer to opt for the self-contained solution.
PGP is a core functionality and it make sense to use the standard implementation.
Here we are using it as a tool for getting the job done.

Reply:
Every dependency we bring into sbt is a long-term liability. here's the transitive dependencies for the current release:

```
$ cs resolve org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r
com.googlecode.javaewah:JavaEWAH:1.1.7:default
com.jcraft:jsch:0.1.55:default
com.jcraft:jzlib:1.1.1:default
org.bouncycastle:bcpg-jdk15on:1.64:default
org.bouncycastle:bcpkix-jdk15on:1.64:default
org.bouncycastle:bcprov-jdk15on:1.64:default
org.eclipse.jgit:org.eclipse.jgit:5.7.0.202003110725-r:default
org.slf4j:slf4j-api:1.7.2:default
```

Note that (not really maintained) sbt-git uses a much older version of JGit. Keeping this in the plugin allows the dependencies to be updated without worrying about bincompat impact etc. Eventually when enough ppl start to use this and realize we need JGit, we can always add more later.